mirascope/base/extractors.py

```diff
@@ -536,3 +538,55 @@ def _setup(
             return_tool = True
         kwargs["tools"] = [tool]
         return kwargs, return_tool
+
+    @classmethod
+    def from_extractor(
+        cls,
+        base_extractor_type: type[BaseExtractorT],
+        call_params: BaseCallParams,
+        *,
+        extract_schema: Optional[ExtractedType] = None,
+    ) -> type[BaseExtractorT]:
+        """Dynamically creates a new extractor class from a base model and an extractor.
```

Comment:
nit: update to "Returns an `extractor_type` instance generated dynamically from this base extractor."